### PR TITLE
Update init_ddp_connection's name and log

### LINKED ISSUE #10256: Update init_ddp_connection logging info message and update function name
## Proposed refactoring or deprecation

init_ddp_connection is not only used by DDP, but also used by FSDP and other plugins inherit DDP. The log info and rank_zero_info saying ""initializing ddp" or "DDP processes", which could be confusing for user

### Motivation

Debuggability and readability 

### Pitch

Update info message to be generic message
Update function name to initialize_distributed_connction as it's used by fsdp and sdp

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?
FSDP and DDP2 inherit DDP and calling setup_distributed() functions, which calls init_ddp_connection. In log, it shows "initial DDP connection" which is confusing. This PR renamed the function and fixed the log

Fixes #10256 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/distributed.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def init_ddp_connection(
     os.environ["MASTER_ADDR"] = cluster_environment.master_address()
     os.environ["MASTER_PORT"] = str(cluster_environment.master_port())
     if torch.distributed.is_available() and not torch.distributed.is_initialized():
-        log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
+        log.info(f"initializing distributed: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
```
Comment (carmocca) [inline pytorch_lightning/utilities/distributed.py:385]:
```suggestion
        log.info(f"Initializing distributed: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
```

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/utilities/distributed.py [hunk 2]
```diff
@@ -360,15 +360,15 @@ def tpu_distributed() -> bool:
     return _TPU_AVAILABLE and xm.xrt_world_size() > 1
 
 
-def init_ddp_connection(
+def init_distributed_connection(
```
Comment (ananthsub) [inline pytorch_lightning/utilities/distributed.py:363]: 1. this is a utility which could be used elsewhere. you'll need to provide backwards compatibility for `init_ddp_connection` - otherwise renaming this is a breaking change.
2. `init_dist_connection` ? slightly less verbose

**Comment by ananthsub** [comment]
proposal to have a `dist` label on github to classify distributed comms/training/execution issues:
right now everything is grouped under the `DDP` label, but that should be reserved for DistributedDataParallel items.

@PyTorchLightning/core-contributors wdyt?

File: pytorch_lightning/utilities/distributed.py (hunk 2 above)
Comment (four4fish) [reply pytorch_lightning/utilities/distributed.py:363]: I didn't see valid use case of using or override init_ddp_connection() in internal fb. Is this BC mainly for who customize TTP?

Comment (ananthsub) [reply pytorch_lightning/utilities/distributed.py:363]: actually, this was added in https://github.com/PyTorchLightning/pytorch-lightning/pull/9044 which was targeted fro v1.5. meaning this code isn't available already, so there's no existing version to be backward compatible with

**Review by ananthsub — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
